src/components/Approval/Approval.tsx

```diff
@@ -157,6 +146,38 @@ export const Approval = () => {
     showErrorToast,
   ])
 
+  const toggleIsExactAllowance = useCallback(() => {
+    setIsExactAllowance(!isExactAllowance)
+  }, [isExactAllowance])
+
+  useEffect(() => {
+    setBuildApprovalTxInput(undefined)
+    createBuildApprovalTxInput(isExactAllowance)
+      .then(buildApprovalTxInput => {
+        setBuildApprovalTxInput(buildApprovalTxInput)
+      })
+      .catch(e => {
+        showErrorToast(e)
+        return history.push(TradeRoutePaths.Input)
+      })
+  }, [isExactAllowance, createBuildApprovalTxInput, history, showErrorToast])
+
+  const approvalNetworkFeeCryptoHuman = useMemo(() => {
+    if (!buildApprovalTxInput) return
+
+    const { gasLimit, gasPrice, maxFeePerGas } = buildApprovalTxInput
+
+    // use EIP-1551 gas where possible
```

Comment:
Do we want to check for `shouldUseEIP1559Fees` ?

Reply:
I dont think its necessary because this is checked upstream and the existence of `maxFeePerGas` implies eip-1559 support. Checking it again here would introduce async to this useMemo which would be more confusing and complex

Reply:
yeah now that we have the wallet available, we can be more accurate with the gas price we are using (legacy vs eip1559)

Reply:
agreed, although this is actually checked inside the helper so im thinking to avoid introducing a second loading state to the UI we can rely on implicit eip-1559 values in the same way as here https://github.com/shapeshift/web/pull/4492#discussion_r1195495300

Reply:
Very true - one or the other being defined will effectively be our `shouldUseEIP1559Fees` check, disregard me!